### PR TITLE
Change ordering of actions

### PR DESCRIPTION
**Pull request by prmshepherd**
Update the ordering of the actions as the one literal on default option is not working for descriminator properties

### CODE REVIEW

**Comment by koxudaxi** [comment]
@prmshepherd
Sorry for my late response.
Thank you for creating the PR.
Could you please provide the test case?
I would like to add the unittest for the changes.

**Comment by prmshepherd** [comment]
Hi @koxudaxi, sorry I have this in draft as more work is needed by me. My use-case is when I am using a discriminator 


```
openapi: 3.0.0
info:
  title: Pet API
  version: 1.0.0
paths:
  /pets:
    post:
      summary: Add a new pet
      requestBody:
        required: true
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/Pet'
      responses:
        '200':
          description: Successfully added the pet
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Pet'
components:
  schemas:
    Pet:
      type: object
      oneOf:
       - $ref: '#/components/schemas/Dog'
       - $ref: '#/components/schemas/Cat'
      discriminator:
        propertyName: petType
        mapping:
          dog: '#/components/schemas/Dog'
          cat: '#/components/schemas/Cat'
    Dog:
      properties:
        breed:
          type: string
        barkVolume:
          type: integer
          format: int32
          description: The volume of the dog's bark
    Cat:
      properties:
        color:
          type: string
        isIndoor:
          type: boolean
          description: Indicates if the cat is an indoor cat
```

This generates:
```
class Dog(BaseModel):
    breed: Optional[str] = None
    barkVolume: Optional[int] = Field(None, description="The volume of the dog's bark")
    petType: Literal['dog']
```

What I would like is `petType: Literal['dog'] = 'dog'`.

But I think the ordering of those actions prevents that

**Comment by koxudaxi** [comment]
@prmshepherd 
Thank you for explaining it 
Could you please add the unittest for the case?

File: tests/data/expected/main/openapi/discriminator/enum_one_literal_as_default.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    v2 = 'v2'
+
+
+class RequestBase(BaseModel):
+    version: RequestVersionEnum
+
+
+class RequestV1(RequestBase):
+    request_id: str = Field(..., description='there is description', title='test title')
+    version: Literal['v1'] = 'v1'
```
Comment (prmshepherd) [inline tests/data/expected/main/openapi/discriminator/enum_one_literal_as_default.py:24]: The generator now correctly adds the default for the literal

File: tests/main/openapi/test_main_openapi.py [hunk 2]
```diff
@@ -2798,3 +2798,31 @@ def test_main_openapi_msgspec_use_annotated_with_field_constraints():
...
+            [
+                '--input',
+                str(OPEN_API_DATA_PATH / 'discriminator_enum.yaml'),
+                '--output',
+                str(output_file),
+                '--input-file-type',
+                'openapi',
+                '--output-model-type',
+                'pydantic_v2.BaseModel',
+                '--use-one-literal-as-default',
```
Comment (prmshepherd) [inline tests/main/openapi/test_main_openapi.py:2817]: Setting --use-one-literal-as-default in the test

**Comment by prmshepherd** [comment]
@koxudaxi Sorry for the delay on this. I've added the test case now

File: tests/main/openapi/test_main_openapi.py (hunk 2 above)
Comment (prmshepherd) [reply tests/main/openapi/test_main_openapi.py:2817]: I'll get onto that failure - not sure how it didn't trigger on my local run

**Comment by prmshepherd** [comment]
@koxudaxi Fixed that test error with the extraneous args - I'm not sure why I didn't see that on my local run.

**Comment by prmshepherd** [comment]
@koxudaxi is anything more needed from me for this one?

**Pull request merged**